cmd/access/node_builder/access_node_builder.go

```diff
@@ -39,6 +39,7 @@ import (
 	"github.com/onflow/flow-go/module/metrics"
```

Comment:
Asking this question here, which has nothing to do this line of code. Just so, we can discuss in this thread.

My question is: What would be the performance implication for this change?

The reason I ask is that, a while ago we [discovered EN's memory was abused](https://github.com/dapperlabs/flow-go/issues/5952#issue-1031456155) from [the report ](https://www.notion.so/dapperlabs/Examining-Leo-s-multi-fork-CPU-spike-issue-f7cecbfb9c3c40409bc258a09f4aa13e#56c2880fd2d64332a198dce3fde5d4bc)that @simonhf made.

Both the top `alloc-space` and `alloc-objects` is `CacheHit`:
```
github.com/onflow/flow-go/module/metrics.(*CacheCollector).CacheHit
```

To fix that, [I disabled the metrics for cache](https://github.com/onflow/flow-go/pull/1518/files).

Now this PR is adding metrics for a different cache, which reminds me that issue. 

I think the reason it was an issue before is that whenever the cache API is visited, we will call into metrics API which somehow allocates objects on heap, as cache API is called frequently, it causes memory abuse.

To mitigate that, maybe we could record metrics less frequently? If we were to record `Size()` of a cache, instead of measuring once for every single call, we could record once for every 100 calls. However, we would have to keep track of the state of total called number, and ensure concurrent-safe, which would be a bit more complex.

Thoughts?

cc @Kay-Zee